crates/kubelet/src/pod/mod.rs

```diff
@@ -153,17 +154,26 @@ impl Pod {
             .find(|container| container.name() == key.name())
     }
 
-    /// Finds the index of the container in the Pod's container status.
+    /// Finds the index of the container in the Pod's container statuses.
     pub fn container_status_index(&self, key: &ContainerKey) -> Option<usize> {
-        let containers: Vec<Container> = if key.is_init() {
-            self.init_containers()
+        let statuses: Vec<KubeContainerStatus> = if key.is_init() {
+            self.kube_pod
+                .status
+                .clone()
```

Comment:
I dont think you can really do this: `.unwrap_or_else(|| &PodStatus::default())`, because the reference doesnt outlive the closure. I will see what I can do though.